NodeGraph: Fix invisible arrow tips in Editor

timo opened the pull request:
**Which issue(s) does this PR fix?**:

When opening the editor on a NodeGraph panel, arrow tips are invisible.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.

timo committed:
NodeGraphPanel: Namespace marker IDs to fix invisible arrow tips in editor

timo commented:
![image](https://github.com/grafana/grafana/assets/25167/523bfb63-4267-4d26-9f39-189b02900b27)
![image](https://github.com/grafana/grafana/assets/25167/b7e6cf3a-27fd-4ce6-a76d-802701f4e4db)
![image](https://github.com/grafana/grafana/assets/25167/30415149-0d4f-4af8-9309-d10be72dab96)

in the third screenshot you can see the inspector jumped to the node that was returned by getElementById, and from the grey coloration you can tell it's the invisible copy of the NodeGraphPanel rather than the one in the foreground that we would have expected

timo commented:
Hi, is it OK to ping someone directly for a review? Is there something more i need to do before someone can look at this? I've seen some commits from you to the Node Graph Panel, @aocenas do you have a little bit of guidance for me on this pull request and the a bit less critical but also even less code-heavy pull request https://github.com/grafana/grafana/pull/84609

timo commented:
> Hey @timo sorry sometimes things fall through the cracks unfortunately so in that cases direct ping is probably good thing. Looking at the comments here can you please repost the screenshots you made? For some reason they show as text links instead of inlined pictures though the URLs are correct.

Hi, it looks like using the <summary> feature broke displaying the images. i removed the summary around them and now they are visible again.

> Maybe we could just generate a UUID for each node graph instead and use that.

Sounds like I would just create a new random UUID in the outermost component's constructor or more likely render function and store it with ... useMemo is more correct than useState I imagine?

aocenas commented:
> Sounds like I would just create a new random UUID in the outermost component's constructor or more likely render function and store it with ... useMemo is more correct than useState I imagine?

I guess we can just use https://react.dev/reference/react/useId here. It talks about accessibility but it is basically about generating a unique ID HTML attribute for the DOM elements which is what we need here. I assume it may be a bit more efficient than generating UUID as it has lesser uniqueness requirements.

timo commented:
Ok, the code uses useId now instead of useMemo + uuidv4

aocenas commented:
thanks yeah looks good but can you run prettier on?
```
[warn] public/app/plugins/panel/nodeGraph/NodeGraph.tsx
[warn] public/app/plugins/panel/nodeGraph/NodeGraphPanel.tsx
```
Otherwise CI won't pass and as this is your main branch I cannot push.

Or just apply this patch
[timo.patch](https://github.com/grafana/grafana/files/14884627/timo.patch)

timo commented:
haha sorry about that, I fumbled the vscode github extension or something and ended up with the code in main, and I couldn't change the branch without closing and creating the pull request anew.

anyway, prettier is done and the code should now be good to merge.

thanks for the assistance!

timo commented:
hate to be a bother, but ping @aocenas ? :)

aocenas commented:
@timo no worries and my apologies, I was sick for quite some time. Seems like there is a problem with the CI though. Should be fixed if you could merge/rebase main. Or you could create a PR from a separate non main branch and give us write access that way I can deal with it.

timo commented:
Ah no worries, you don't owe me nothing, just making sure nothing gets lost :)

I squashed the commits down to one since the back-and-forth wasn't worth much, and rebased on latest upstream main, hope it works :)

timo commented:
ok it looks like i really have to recreate the pull request from a different branch or something?

I don't think we can see what $DRONE_PULL_REQUEST is set to in the drone continuous integration run to find out why that check for the head commit's repo doesn't have the "fork" set to "false" - probably the URL that it's curling has nothing where the pull request number is supposed to go and that gives something entirely different??

what the drone ci log shows:
```
+ is_fork=$(curl "https://$GITHUB_TOKEN@api.github.com/repos/grafana/grafana/pulls/$DRONE_PULL_REQUEST" | jq .head.repo.fork)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 21791  100 21791    0     0  84616      0 --:--:-- --:--:-- --:--:-- 84789
+ if [ "$is_fork" != false ]; then return 1; fi
```

what i see when i curl the data for this exact pull request without a github api token:

```
curl "https://api.github.com/repos/grafana/grafana/pulls/84683" > /tmp/grafana_pull.json  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 35060    0 35060    0     0  68625      0 --:--:-- --:--:-- --:--:-- 68745
```

so it's about 16 kilobytes shorter for some reason

aocenas commented:
@timo thanks for the separate PR at least it should be possible for me to push there if needed. I think the problem is somewhere on our end, external PRs have some specifics for CI because of security reasons, so some steps are expected to fail and that should not fail the whole build. Here it seems like some swagger API def generation fails and I don't really know why at this moment trying to figure it out just to make sure it isn't going to block other external PRs.
